packages/ifpack2/src/Ifpack2_Relaxation_def.hpp

```diff
@@ -362,12 +372,23 @@ void Relaxation<MatrixType>::setParametersImpl (Teuchos::ParameterList& pl)
 
   Teuchos::ArrayRCP<local_ordinal_type> localSmoothingIndices = pl.get<Teuchos::ArrayRCP<local_ordinal_type> >("relaxation: local smoothing indices");
 
+  // for Two-stage Gauss-Seidel
+  if (pl.isType<double>("relaxation: inner damping factor")) {
```

Comment:
Maybe change this to
```
if (!std::is_same<double, ST>::value && pl.isType<double>("relaxation: inner damping factor")) {
```
and move it before
```
const scalar_type innerDampingFactor = STS::one ();
pl->set ("relaxation: inner damping factor", innerDampingFactor, "Damping factor for the inner sweep of two-stage GS");
```
I'd hope that it would help avoid having to change the MueLu gold files. Cause MueLu prints "relaxation: inner damping factor" since it's not default anymore.

Reply:
Thank you, @cgcgcg. Something like this? Now, I see "[default]" after the print.

Reply:
@cgcgcg Thanks.  The main issue was that the gold files for the ML parameter list conversion tests.

Reply:
@jhux2 Unless I messed up, all MueLu interface tests should be ignoring default parameters. Before @iyamazaki change, "relaxation: inner damping factor" wasn't default, so it was getting checked. And I assume the gold files for this PR were updated without ML. With this change, you shouldn't have to update the gold files at all.

Reply:
@iyamazaki Wait, you got [default] even though you didn't move those line before `pl.validateParametersAndSetDefaults (* getValidParameters ())`?

Reply:
Hi, @cgcgcg. "[default]" wasn't there before, but now I see them.

Reply:
Ok, let's see what the AT says. I'm hoping it will fail, since we now don't need the updated gold files.

Reply:
Should I revert all the gold files?

Reply:
@iyamazaki Looks like I messed up somewhere. I'll try to fix it, and then update your branch.

Reply:
I fixed the MueLu issue in PR #9011. Once that is merged, this PR won't need any gold file changes. Just tested that locally. In order to avoid merge conflicts, I'd outright nuke commit feea4d8712955835a42fa124108541995274b002 and force push.

Reply:
Thank you, @cgcgcg. I just nuked that gold files.